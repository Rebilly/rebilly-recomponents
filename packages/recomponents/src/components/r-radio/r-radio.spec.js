import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RRadio from './r-radio.vue';

describe('r-radio.vue', () => {
    it('renders default radio correctly', () => {
        const label = `label-${new Date().getTime()}`;

        const wrapper = shallowMount(RRadio, {
            propsData: {
                label,
                checked: '',
                value: '1',
            },
        });

        expect(wrapper.find('label').text()).toEqual(label);
        expect(wrapper.contains('.r-field-caption')).toBe(false);
    });

    it('should render caption props if passed', () => {
        const caption = `caption-${new Date().getTime()}`;

        const wrapper = shallowMount(RRadio, {
            propsData: {
                caption,
                checked: '',
                value: '1',
            },
        });

        expect(wrapper.find('.r-field-caption').text()).toEqual(caption);
    });


    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RRadio, {
            propsData: {
                checked: 'value',
                value: 'value',
                label: 'Checkbox',
                name: 'radio-group',
                id: 'id1234',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', () => {
        const wrapper = renderToString(RRadio, {
            propsData: {
                checked: 'value',
                value: 'value',
                label: 'Checkbox',
                name: 'radio-group',
                id: 'id1234',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should be invalid if validation is dirty', () => {
        const wrapper = mount(RRadio, {
            propsData: {
                checked: 'value',
                value: 'value',
                label: 'Checkbox',
                validate: {
                    $invalid: true,
                    $dirty: false,
                },
                name: 'radio-group',
                id: 'id1234',
            },
        });

        expect(wrapper.isInvalid).toBeFalsy();
    });
});
