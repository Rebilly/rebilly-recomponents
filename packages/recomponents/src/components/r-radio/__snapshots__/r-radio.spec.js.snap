// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-radio.vue should render Wrapper and match snapshot 1`] = `
<div role="radio" aria-checked="true" class="r-field"><label for="id1234" class="r-field-label r-field-label-toggle is-checked">
    Checkbox
  </label> <input type="radio" id="id1234" name="radio-group" class="r-field-radio" value="value"> <span class="r-field-radio-style"><r-icon-stub icon="radio-checkmark"></r-icon-stub></span>
  <!---->
</div>
`;

exports[`r-radio.vue should render via SSR and match snapshot 1`] = `
<div role="radio" aria-checked="true" class="r-field"><label for="id1234" class="r-field-label r-field-label-toggle is-checked">
    Checkbox
  </label> <input type="radio" id="id1234" name="radio-group" value="value" checked="checked" class="r-field-radio"> <span class="r-field-radio-style"><svg class="r-icon r-icon-20"><use xlink:href="#icon-radio-checkmark" xmlns:xlink="http://www.w3.org/1999/xlink"></use></svg></span>
  <!---->
</div>
`;
